src/string.js

```diff
@@ -35,3 +36,17 @@ export let autoLabelOption = a => ({
   label: a.label || autoLabel(a.value || a),
 })
 export let autoLabelOptions = _.map(autoLabelOption)
+
+export let toSentenceWith = _.curry((initial, last, join, array) =>
+  _.flow(
+    mapIndexed((x, i) => [
+      x,
+      i === array.length - 1 ? null : i === array.length - 2 ? last : initial,
+    ]),
+    _.flatten,
+    _.compact,
+    join || _.identity
+  )(array)
+)
```

Comment:
So all together I think it would be something like this:

```js
export let toSentenceWith = _.curry((separator, lastSeparator, array) =>
  _.flow(
    F.mapIndexed(
      (x, i) =>
        x +
        (i < array.length - 1
          ? i < array.length - 2
            ? separator
            : lastSeparator
          : '')
    ),
    _.join('')
  )(array)
)
```

Reply:
It's not what I'm intending to do with this function! See my comment here: https://github.com/smartprocure/futil-js/pull/246#discussion_r222250673